Delete request

scmountain opened the pull request:
@neight-allen @case-eee @BobGu:

We have a mysterious issue in our server-test.js file. Testing on the whole has been causing us problems. On line 115 you will see we had to reestablish our this.request(as close_request) due to the fact our this.request would always be undefined. While the hack Bob helped us with works, we're still wondering why this is happening? We've triple-checked the }); brackets and they're all lined up.

You'll also notice that 131-135 is commented out. This afterEach was giving me an unresolved promise hook error. In a similar vein, our "/foods" endpoint's test will not pass if an afterEach "TRUNCATE" is included below the beforeEach... the endpoint test will return an empty array, meaning the afterEach runs before the test assertions. This doesn't make sense... shouldn't the afterEach run after each "it" block? 

The code for the "delete" functionality looks correct and we have had many peers take a look at it. We're having trouble getting our tests to run properly... and our suspicions point to issues with beforeEach and afterEach interacting with the database. 

I understand it is Saturday, but if any of you get the chance to take a look and maybe find our error, this would really help Max and I move on!

maxglassie committed:
POST request

Building out the post request. There is some issue with the test not
registering the this.request (it returns undefined) so we added a little
hack into the test until we can sort it out. Currently building out the
food controller with an insert method.

maxglassie committed:
insertFood method

The insertFood method works, it returns something. My problem is determining
whether or not it is successful - it doesn't necessarily confirm that -
in order to throw an error if it wasn't and send a positive response if it did.

scmountain committed:
Merge branch 'post_request' of https://github.com/scmountain/Quantified-Self

scmountain committed:
fixed merge conflicts

scmountain committed:
WIP issues with server-test.js

scmountain committed:
found the missing done in the delete function

after finding the food, we are now nested inside that if block
therefore we needed to add another .then and done to close of the delete request
inside the if block

case-eee commented on lib/controllers/foods-controller.js:
```diff
@@ -26,4 +26,39 @@ const FoodsRouter = require('express').Router();
+      Food.find(request.params.id)
+        .then((data) => {
+          if (data) {
+            Food.updateFood(data.id)
```
@scmountain @maxglassie i believe this `Food.updateFood(data.id)` will need a `then` attached for it to execute. i know this is in your `put` request, but it could be causing problems

scmountain committed:
tests delete and post, put method is lacking

delete tests and has been confirmed on local host. POST method now not only adds to the DB,
but also returns the item that was created.
the test for put needs to have an object passed to it, to test

maxglassie commented on lib/controllers/foods-controller.js:
```diff
@@ -18,12 +18,54 @@ const FoodsRouter = require('express').Router();
-      .then( data => {
-        response.sendStatus(201);
+    .then( data => {
+        response.send(data);
```
Thanks for updating this to meet conventions; this will help our front end to have the data be returned in the POST request.

scmountain replied:
makes sense too, want to see what we create!

scmountain committed:
Fixes merge conflicts

mergeing of both locals, and fixes all merge conflicts, PUT is still unfinished

maxglassie commented on test/server-test.js:
```diff
@@ -129,7 +161,7 @@ describe("Server", function(){
           });
 
-      it('should not be a 404 error', (done) => {
+      it('should not be a 404 error', function(done) {
```
Thanks for digging into this and translating back to ES5.

maxglassie commented:
I pulled the branch, ran the tests, checked it in the browser, and examined the code. Good job Spencer.
